lib/common/http1client.c

```diff
@@ -608,7 +634,9 @@ static void req_body_send_complete(h2o_socket_t *sock, const char *err)
 
     int is_end_stream = client->body_buf.is_end_stream;
 
-    call_proceed_req(client, NULL);
+    if (client->proceed_req != NULL) {
+        call_proceed_req(client, NULL);
+    }
```

Comment:
This looks like a change to the state machine. Could you please elaborate how you found this to be an issue and why you think this is the correct way to fix?

Reply:
This is actually not changing state machine, but is just reusing the existing code. Before this PR, `req_body_send` was only used when req streaming is enabled, but this PR utilizes that function also in the case of non-streaming case. In that case the client doesn't have `proceed_req`, so we need this check to support both cases.